tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,197 @@
+use tokio::process::Command;
+
+/// Takes a local kit built using buildsys and publishes it
+/// to a vendor specified in Infra.toml
+#[derive(Debug, Parser)]
+pub(crate) struct PublishKitArgs {
+    /// Location of the local kit
+    #[arg(long)]
+    kit_path: PathBuf,
+    /// Vendor to publish kit to
+    #[arg(long)]
+    vendor: String,
+}
+
+macro_rules! docker {
+    ($arg: expr) => {{
+        let result = Command::new("docker")
+            .args($arg)
+            .output()
+            .await
+            .context(error::IoSnafu)?;
+        ensure!(
+            result.status.success(),
+            error::PublishFailSnafu {
+                reason: String::from_utf8_lossy(&result.stderr)
+            }
+        );
+        result.stdout
+    }};
+}
+
+pub(crate) async fn run(args: &Args, publish_kit_args: &PublishKitArgs) -> Result<()> {
+    // If a lock file exists, use that, otherwise use Infra.toml
+    let infra_config = InfraConfig::from_path_or_lock(&args.infra_config_path, false)
+        .context(error::ConfigSnafu)?;
+    trace!("Parsed infra config: {:?}", infra_config);
+    ensure!(infra_config.vendor.is_some(), error::NoVendorsSnafu);
+
+    // Fetch the vendor container registry uri
+    let vendor = infra_config
+        .vendor
+        .as_ref()
+        .unwrap()
+        .get(&publish_kit_args.vendor)
+        .context(error::VendorNotFoundSnafu {
+            name: publish_kit_args.vendor.clone(),
+        })?;
+    let vendor_registry_uri = vendor.registry.clone();
+    debug!(
+        "Found vendor container registry at uri: {}",
+        vendor_registry_uri
+    );
+
+    // Auto resolve the expected paths for the kit contents archive
+    let kit_path = publish_kit_args.kit_path.as_path();
+    let kit_name = kit_path.file_name().unwrap().to_string_lossy();
+
+    let kit_version = find_version(kit_path, &kit_name)?;
+
+    let mut platform_images = Vec::new();
+    for arch in ["aarch64", "x86_64"] {
+        let kit_filename = format!("{}-{}-{}.tar", &kit_name, &kit_version, arch);
+        let path = kit_path.join(&kit_filename);
+
+        let out = docker!(["load", format!("--input={}", path.display()).as_str(),]);
+        let out = String::from_utf8_lossy(&out);
+        let digest_expression =
+            Regex::new("(?<digest>sha256:[0-9a-f]{64})").context(error::RegexSnafu)?;
+        let caps = digest_expression.captures(&out).unwrap();
+        let digest = &caps["digest"];
+
+        let arch_specific_target_uri = format!(
+            "{}/{}:{}-{}",
+            vendor_registry_uri, kit_name, &kit_version, arch
+        );
+
+        docker!(["tag", digest, &arch_specific_target_uri,]);
+
+        info!(
+            "Pushing kit image for platform {} to {}",
+            arch, &arch_specific_target_uri
+        );
+        docker!(["push", &arch_specific_target_uri,]);
+
+        let docker_arch =
+            DockerArchitecture::try_from(arch).context(error::InvalidArchitectureSnafu)?;
```

Comment:
nit: can do this at the outset so we don't do any `docker` commands if this is just going to fail

(we know it's infallible though)